### PR TITLE
Unify the version naming convention

### PR DESCRIPTION
**Pull request by mgol**
So far some browsers were just omitting a dot in the version string and some
replaced it by an underscore. This commit changes all those versions to use
the underscore translation. This will reduce confusion; with some environments
using rapidly increasing major version numbers it becomes hard to distinguish
"41" from "4.1" when both are translated to "41" in the browser name.

The only exception is Duktape which needs to have names maching keys matching
what Duktape.version reports.

Also:
### 1)
Rename Duktape shortcuts from "DUK 20" to "DUK 2.0" & "DUK 21" to "DUK 2.1"

The previous versions were incorrect. Lots of engines have high major version
numbers so version 21 sounded plausible.

### 2)
Change iOS 10.0-10.3 to 10.0-10.2

iOS 10.3 was incorrectly included in a group with 10.0-10.2.

### CODE REVIEW

**Review by ljharb — APPROVED** [review]
We should fix the duktape result generation as well.

**Comment by mgol** [comment]
> We should fix the duktape result generation as well.

This means there will be no direct matching between the key we use and the one reported by `Duktape.version`. Is that OK?

**Comment by ljharb** [comment]
`Duktape.version` reports `duktape20`? O.o

**Comment by mgol** [comment]
Unfortunately, yes...

**Comment by mgol** [comment]
Hmm, no, actually it seems the comment is not 100% correct. What it actually reports is:
```
duk> Duktape.version
= 20100
```
Even weirder but then if we already have to transform it we might as well add an underscore.

**Comment by ljharb** [comment]
yeah, that's a really stupid version string; it should use semver. I see no value in trying to match that.

**Comment by mgol** [comment]
It's not even a string, it's a number. :)

**Comment by mgol** [comment]
@ljharb Should I change that script in this PR and unify Duktape entries as well? Or in a new one?

**Comment by ljharb** [comment]
Let's do another commit in this pr

**Comment by mgol** [comment]
@ljharb PR updated.

**Review by ljharb — APPROVED** [review]

File: environments.json [hunk 1]
```diff
@@ -1265,7 +1265,7 @@
       "non-standard"
     ]
   },
-  "safari51": {
```
Comment (afmenez) [inline environments.json]: This will conflict with my patch on #1099

File: environments.json [hunk 2]
```diff
@@ -1283,7 +1283,7 @@
     "short": "SF 6.1,<br>SF 7",
     "obsolete": true
   },
-  "safari71_8": {
+  "safari7_1_8": {
```
Comment (afmenez) [inline environments.json]: This is not Safari 7.1.8, but Safari 7.1 & Safari 8.  Also fixed on my patch on #1099

**Review by afmenez — CHANGES_REQUESTED** [review]
It would be really nice if you could revert the Safari changes so we can merge my more complete patch for Safari (#1099) without conflicts.

**Comment by mgol** [comment]
@afmenez You mean Safari & iOS changes, don't you?

@ljharb Does that sound OK to you? I can revert those particular changes.

**Comment by afmenez** [comment]
Yep, I did a larger change on Safari and iOS, adding some versions and results, and it already included all the changes you did.  If you revert just these parts, I believe both of our patches will be able to merge.  :)

File: environments.json (hunk 1 above)
Comment (ljharb) [reply environments.json]: Do you think it'd be easier to merge yours first, and then rebase this one? or would it be easier to merge this, and rebase yours on top?

Comment (afmenez) [reply environments.json]: I believe both would be about the same amount of work.  But I pushed my patch first... :)

Comment (mgol) [reply environments.json]: @ljharb I can wait for the @afmenez's patch to be landed & then rebase mine. I'm fine with solving those merge conflicts, I just hope I won't have to do that multiple times.

Comment (ljharb) [reply environments.json]: ok, sounds good

**Comment by mgol** [comment]
PR rebased.

**Pull request merged**